index.js

```diff
@@ -373,6 +373,9 @@ class Peer extends stream.Duplex {
   }
 
   negotiate () {
+    this._debug('negotiate')
+    this.emit('negotiate')
```

Comment:
Unrelated fix here: I moved this debug log and the undocumented `'negotiate'` event to fire when negotiate happens. Before, it was firing when the connection becomes stable, which seemed like a bug

Reply:
It's meant to fire when negotiation completes, which coincides with `stable` without a queued renegotiation.

It's only used in some tests.

Reply:
I think it'd be better to name the event something other than `'negotiate'` since the convention is to emit debug logs and events named after methods when they're first called. Maybe it should be called `'negotiated'` then?